### PR TITLE
Update Gradle build version

### PR DESCRIPTION
**Pull request by loay**
connect to https://github.com/strongloop-internal/scrum-loopback/issues/760

### CODE REVIEW

**Comment by loay** [comment]
connect to https://github.com/strongloop-internal/scrum-loopback/issues/760

File: build.gradle [hunk 1]
```diff
@@ -5,14 +5,14 @@ buildscript {
       mavenCentral()
   }
   dependencies {
-      classpath 'com.android.tools.build:gradle:1.1.0'
-      classpath 'org.apache.commons:commons-io:1.3.2'
-      classpath 'com.github.dcendents:android-maven-plugin:1.2'
+      classpath 'com.android.tools.build:gradle:1.5.+'
+      classpath 'commons-io:commons-io:2.4'
+      classpath "com.github.dcendents:android-maven-gradle-plugin:1.3"
```
Comment (bajtos) [inline build.gradle]: Please keep single quotes for consistency, check & fix other places in this diff too please.

File: build.gradle [hunk 2]
```diff
@@ -33,13 +33,17 @@ repositories {
...
-
+    compileSdkVersion 22
+    buildToolsVersion "22.0.1"
     defaultConfig {
         minSdkVersion 8
         targetSdkVersion 21
     }
+    compileOptions {
+        encoding "UTF-8"
+        sourceCompatibility JavaVersion.VERSION_1_7
```
Comment (bajtos) [inline build.gradle]: I think we were initially targeting Java version 1.6, but if 1.7 works too then I am fine with this too.

File: build.gradle [hunk 3]
```diff
@@ -254,13 +258,19 @@ afterEvaluate { project ->
...
       }
     }
-    compileReleaseJava.dependsOn printCompilerInput
+    compileReleaseJavaWithJavac.dependsOn printCompilerInput
 }
+
+allprojects {
+    repositories {
+        mavenCentral()
+    }
```
Comment (bajtos) [inline build.gradle]: This seem to duplicate L27-L33, please double check and try to keep the list of repositories used to fetch project dependencies in one place only.

Note there is another configuration for build dependencies on L4-L6. I don't know if it's possible to unify them with project dependencies, let's leave that out of scope of current changes.

**Comment by bajtos** [comment]
@loay reviewed, see the comments above. Please check why the CI build is failing and fix.

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@bajtos 
Test Pass now. 
Also, kept single quotation everywhere, except where it would give an error if switched to single.

File: build.gradle [hunk 4]
```diff
@@ -254,13 +258,13 @@ afterEvaluate { project ->
     // debugging
     task printCompilerInput << {
       logger.info "**JAVAC INPUT FILES**"
-      compileReleaseJava.source.each {
+        compileReleaseJavaWithJavac.source.each {
```
Comment (bajtos) [inline build.gradle]: Please revert the indentation change, keep the same indentation level as before.

File: build.gradle [hunk 5]
```diff
@@ -254,13 +258,13 @@ afterEvaluate { project ->
...
         logger.info '  ' + it.absolutePath
       }
       logger.info "**JAVAC CLASSPATH**"
-      compileReleaseJava.classpath.each {
+        compileReleaseJavaWithJavac.classpath.each {
```
Comment (bajtos) [inline build.gradle]: Please revert the indentation change, keep the same indentation level as before.

**Comment by bajtos** [comment]
Two more comments, see above. You can land this patch without waiting for another review once you address them. Thank you!

**Pull request merged**